src/connectionManager.ts

```diff
@@ -11,13 +14,39 @@ export interface Connection {
 }
 
 export class ConnectionManager {
+  private static connectionManager: ConnectionManager;
   private connections: Record<string, Connection>;
   private currentConnection?: Connection;
+  private static _extContext: ExtensionContext;
 
-  constructor() {
+  private constructor() {
     this.connections = {};
   }
 
+  public static getManager() {
+    return ConnectionManager.connectionManager;
+  }
+
+  public static _initialize(ec: ExtensionContext) {
+    if (ConnectionManager.connectionManager === undefined) {
```

Comment:
This is cleaner as an early return:

```javascript
if (ConnectionManager.connectionManager !== undefined) {
  return;
}
```

Or more concisely,

```javascript
if (!ConnectionManager.connectionManager) {
  return;
}
```

See my comment on reducing nesting below.